apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/CorrelationContext.java

```diff
@@ -17,26 +17,38 @@
 
 package org.apache.skywalking.apm.agent.core.context;
 
-import org.apache.skywalking.apm.agent.core.base64.Base64;
-import org.apache.skywalking.apm.agent.core.conf.Config;
-import org.apache.skywalking.apm.util.StringUtil;
-
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.stream.Collectors;
+import org.apache.skywalking.apm.agent.core.base64.Base64;
+import org.apache.skywalking.apm.agent.core.conf.Config;
+import org.apache.skywalking.apm.agent.core.context.tag.StringTag;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.util.StringUtil;
+
+import static org.apache.skywalking.apm.agent.core.conf.Config.Correlation.AUTO_TAG_KEYS;
 
 /**
- * Correlation context, use to propagation user custom data.
- * Working on the protocol and delegate set/get method.
+ * Correlation context, use to propagation user custom data. Working on the protocol and delegate set/get method.
  */
 public class CorrelationContext {
 
     private final Map<String, String> data;
 
+    private final List<String> autoTagKeys;
+
     public CorrelationContext() {
         this.data = new HashMap<>(Config.Correlation.ELEMENT_MAX_NUMBER);
+        if (StringUtil.isNotEmpty(AUTO_TAG_KEYS)) {
+            autoTagKeys = Arrays.asList(AUTO_TAG_KEYS.split(","));
+        } else {
```

Comment:
From my understanding, this should be in the `static{}` block, as it doesn't need to execute every time.